plenum/server/replica.py

```diff
@@ -1883,8 +1883,9 @@ def markCheckPointStable(self, seqNo):
         for k in previousCheckpoints:
             self.logger.trace("{} removing previous checkpoint {}".format(self, k))
             self.checkpoints.pop(k)
-        self._remove_stashed_checkpoints(till_3pc_key=(self.viewNo, seqNo))
-        self._gc((self.viewNo, seqNo))
+        view_no = self.last_ordered_3pc[0]
```

Comment:
Can we mark checkpoint stable if we don't order anything? In this case last_ordered can be much below than the stable checkpoint's seqNo.

Reply:
Is this case reproduce then node in catchup?

Reply:
We clean 3pc messages for transactions ordered in catchup  in the method _remove_till_caught_up_3pc in catchup 
 https://github.com/hyperledger/indy-plenum/blob/3641dafe6fadf6528c2dfe3f5938d67bfe383bef/plenum/server/replica.py#L2557-L2579